### ISSUE TITLE
FT.AGGREGATE allow with score and return always 0

### ISSUE THREAD

**Issue by MeirShpilraien** [RediSearch/RediSearch#1151]
```
127.0.0.1:6379> ft.search idx foo WITHSCORES
1) (integer) 1
2) "doc1"
3) "1"
4) 1) "f"
   2) "foo"
127.0.0.1:6379> ft.aggregate idx foo WITHSCORES
1) (integer) 1
2) "0"
3) (empty list or set)
```
maybe when using WITHSCORE with ft.aggregate it should add the scorer as the first step and add it as a field to the result?

**Comment by mnunberg** [comment]
We don't calculate the score on FT.AGGREGATE because there is no default sorting order. That is probably the main functional difference between FT.SEARCH and FT.AGGREGATE. I suppose it's possible to add scores if `WITHSCORES` is requested, _and_ no groupby is applied.

**Comment by mnunberg** [comment]
I think the real issue is what kind of query would someone run with FT.AGGREGATE that would still have the scores; and why such a query could not be run on FT.SEARCH. The only thing that comes to mind is `APPLY` and `FILTER` operations, both of which can be used on `FT.SEARCH` (it's more a matter of documentation); and maybe disable `WITHSCORES` on aggregate.

**Comment by MeirShpilraien** [comment]
Someone might want to average score on some groupby reducer .. adding the score as another field seems like the correct way. It will allow to continue using ft.aggregate as search and also as aggregate

**Comment by mnunberg** [comment]
The syntax extensions to 2.0 (which can be ported to 1.6) will have support for `$doc_score` and `$result_score` attributes. In this case, the user can simply do `APPLY $doc_score AS doc_score`

**Comment by MeirShpilraien** [comment]
Yes but you might do not want to calculate it and when you reached the apply it might be too late to calculate it (you might already did a groupby operation) so WITHSCORES can tell you whether or not to calculate the score before start executing the aggregation chain.

**Comment by mnunberg** [comment]
If you did a GROUPBY operation before APPLY $score, then that's a logic error. You can have multiple `APPLY` statements.

For example:

```
FT.AGGREGATE idx match-expr APPLY "$doc_score" AS doc_score GROUPBY 1 @company REDUCE FIRST_VALUE 3 BY @doc_score DESC APPLY somefunc(@company) as company
```

**Comment by MeirShpilraien** [comment]
@mnunberg if you spot this logic error and return error message then this api is fine.
